### PR TITLE
startup: descriptor manifest beside each .das_module, exe require-resolver rows only under the whole-lib link, JIT global lookup memory(none) at the use site, computed goto as one switch

### PR DESCRIPTION
**Pull request by borisbat**
**Behavior change: the module scan writes a `.das_module.manifest` beside every `.das_module` it compiles, and `LLVM_JIT_CODEGEN_VERSION` is 0x75 (0x73-0x75 here, above master's 0x72 fast-math bump) - a warm `-jit` run re-emits its DLL cache once.**

**Why.** A hello world took 61 ms interpreted, 250 ms under `-jit`, and 8.6 ms as a standalone exe, against 1 ms for the same program in zig; the exe spent half its time registering 606 require-resolver rows it could never use, the interpreter spent 19 ms compiling 28 descriptor scripts on every start, and a JIT loop reading a global four times got four opaque base pointers.

**What changes.**
- A standalone exe carries the host's `register_native_path` rows only under the whole-lib link, emitted once; a runtime-only exe carries none.
- The module scan records what each descriptor registers into `.das_module.manifest` next to it and replays that on later starts; the key is the descriptor's content, the content of every module its compile read, its folder, the das root, the binary kind and the cross-compile target; `no_manifest()` inside a descriptor opts it out; descriptors compile under `ignore_shared_modules`, so the program's module set no longer depends on whether the scan compiled.
- The JIT's global-offset lookup is `memory(none)` and emitted at the use site, so LLVM dedups and hoists it and never runs it for an untaken branch.
- A computed goto lowers to one `switch`.
- LINT010 stands down in a function with a label or goto, the way LINT023 does.

**Observable behavior.**
- hello world exe: 8.6 ms, 100 KB -> 4.4 ms, 51 KB; spectral-norm exe 63 -> 59 ms.
- hello world interpreted: 61 ms -> 43 ms (M5); a second start prints `replayed N row(s)` under `DAS_TRACE_MODULE_LOAD=1`.
- A JIT function reading one global N times: N hash probes per call, N base pointers -> one probe, one base; `if/else` writing two globals -> one probe over a `select`.
- An existing module cache written by a scanning start takes one FALLBACK rewrite under this binary (`daslib/builtin` now leads every record set), then serves clean.
- `var res = 10; goto label 8` no longer reports a LINT010 dead store.

**Where to look.** `src/ast/dyn_modules.cpp` (manifest read, replay, write - the format rules are in `src/ast/ARCHITECTURE.md` sec.2), the recorder in `module_builtin_fio.cpp`, `visitExprVar` in `llvm_jit.das`, and `inject_main` in `llvm_exe.das` where the row emission moved below the `needs_whole_lib` decision.

<details>
<summary>Validation, claims, ledger</summary>

### Validation
- Local-only: dasLLVM module suite 84/84 under `-jit`; `tests/language` 1729/1729 interp and `-jit`; `tests/jit_tests` 388/388; `tests/module_cache` 23/23; `exe_paths_module_resolve` ctest; the dasLLAMA `test_exe_smoke` (SmolLM2, real model) 2/2; `dastest.exe` rebuilt as a whole-lib exe still runs a core test and a dasLLVM test with its 303 rows.
- The `.md` review's rulings: the SDK pre-generation promise is out of `src/ast/ARCHITECTURE.md` (the key's root lines are absolute paths); the default-path cwd write is ledgered in `tests/module_cache/ARCHITECTURE.md`; the five tune tests' `-jit` children take `-jit-no-cache` so the new dasLLVM child-artifact rule holds on the suite it binds.
- CI round one: the darwin lint lane read LINT027 on `modules/dasLLVM/ARCHITECTURE.md` (303 lines with the sidecar ledger); the ledger is a fact about the tests folder and now opens `modules/dasLLVM/tests/ARCHITECTURE.md`. Local preflight linted the changed `.das` set only, so the `.md` line gate was first seen in CI; the preflight lint gate now reads changed `.md` files too (`is_lint_input`, tested), and the skill's lint row says so.
- Rebased onto master after its 0x72 fast-math codegen bump landed; the three JIT bumps here renumber to 0x73-0x75 above it. After the rebase: sync, stamp-reach and jit-smoke green, the emitter pin (`tests-cpp-small`) green, `tests/module_cache` and the dasLLVM IR tests green on the rebuilt binary, lint and review-md green.
- Dupes report: `das_exe` / `report_child` in the new manifest test are exact matches of the three sibling tests in `tests/module_cache/` - the folder's per-file child-spawn harness, kept as the folder has it.
- Cache stability, by hand: an explicit `-module-cache` written on a cold-scan start reads `deser: clean` on a warm-scan start.
- Negative controls run: `llvm_jit_global_lookup` reds on the committed emitter (8 lookups, 2 hoisted constants); `llvm_exe_native_paths` reds before the gate (606 rows); `test_lint010_goto` reds on the old lint.das.
- Linux: the module cache and the JIT DLL cache verified on the daslinux arm64 VM (master); the manifest itself was not run on Linux or Windows locally - CI's lanes are the first run there.
- External review rounds (codex): round one, a P1 - the recorder's process-global state under two concurrent scans - fixed as thread-local. Round two on the fixed tip, a P1 and a P2, both real and both fixed here: a descriptor's `require ./helper` was outside the manifest key, so editing the helper replayed stale rows (now every file the descriptor's compile read is a `dep` stamp line, format version 2; a test arm edits the helper and reads `compiled (dependency changed)`); and a `dm` row's `on_error` went through `atoi`, so a damaged field replayed as Quiet (now one of the three `RegisterOnError` values or damage, two damage-table rows). Round three on that tip: one P1, rejected - it claimed the solid-context `g2v` cache leaks a function's entry-block value into the globals-init function, but `build_global_variables` constructs its own `LlvmJitVisitor`, so the cache is empty there; a `solid_context` probe whose initializers read two earlier globals runs clean under `-jit` and the verifier.
- Style-hygiene audit: the `is_dll_build` copy of `das_is_dll_build` dropped, the `g2v` doc corrected to the solid-context arm it serves, restating comments trimmed, `declared_after_jump` renamed to what it is; the two skill-level guide findings (a `//!` that restates an un-anchored architecture section, `//!` on statements) are proposals for `skills/comment_style_hygiene.md`, not applied here.
- Checklist audits found two real defects on the way in, both fixed here: the manifest key missed `get_cross_platform_name()` (dasOpenGL's descriptor reads `--jit-target` from argv) and `get_das_root()` (the examples descriptor reads it) - now key lines, with a test arm; and `LLVM_JIT_EMITTER_HASH` was stale for the edited emitter sources (`tests-cpp-small` emitter pin reds without it).
- The tdd audit (23 branches, negative controls run) named 12 untested; 10 got tests here: every `read_manifest` rejection is a table row (which found the empty-line arm unreachable - `split_tabs` never returns an empty vector - fixed), the `dm` row leg, a descriptor that fails to compile, the tab-in-argument and cannot-create / cannot-rename write paths, the LINT010 label-alone and declared-under-label arms (each red under its mutation), the JIT invalid-label trap (`tests/jit_tests/labels.das` un-comments its case), the whole-lib exe's row count pinned to the host's, and a solid-context twin of the global-lookup fixture. Left untested by design: the use-site emission in isolation (invisible once `memory(none)` lets LLVM fold the calls - the base-code control is red on the combined change) and the `src == nullptr` arm.

### Claims - stated, not tested
- The `.tmp`-then-rename writer on Windows removes the old manifest before the rename; a crash between the two leaves no manifest, which the next start treats as missing. (The cannot-create and cannot-rename arms run on every lane; the crash window itself is not exercised.)
- The Windows arm of `llvm_exe_native_paths.das` (`set "PATH=..."&&` through `_popen`) has not run on a Windows host; the dasLLVM suite has no CI lane, so a Windows run of that one file is owed before the release build.
- The two re-pinned served counts in `test_default_cache_path.das` (`2 module(s)`, `1 module(s)`) were reproduced on macOS only; the record set is builtin + leaves + root on every platform by construction (`ignore_shared_modules` is unconditional), and CI's Linux and Windows lanes run the file.

### Not done
- Lazy dlopen of C++ modules on require miss (the remaining ~30 ms of the interpreter's scan; the ordering contract is in `src/ast/ARCHITECTURE.md` sec.2's neighbours: ctor-time `Module::require`, the `initDependencies` fixed point, DT_NEEDED retry, ctor TypeDecls on the thread root until `Initialize`).
- The `-jit` warm start still deserializes the 34 MB emitter graph on a DLL-cache hit (~140 ms); a hit fast path needs only the bindings record.
- LLVM vectorizes spectral-norm's inner loop 2-wide with an in-order reduction and loses 2x to the scalar form; `[vectorize_disable]` on the loop is the workaround the dasProfile copy already carries.
- A startup table for dasProfile (hello world + every test, exe size, interp / jit / exe, plus a zig lane) as a follow-up once this ships in the SDK.

</details>

### CODE REVIEW

File: modules/dasLLVM/daslib/llvm_jit_common.das [hunk 1]
```diff
@@ -808,9 +812,9 @@ def public init_jit_state(cg_opt_level : uint; target_triple : string; host_feat
         jit_fn_type($(ctx : void?; nArgs : int; types, at, args : void?) : void? {}),
         get_jit_string_builder_temp(), fixed_array(nounwind, willreturn))
     jit_add_extern(FN_JIT_GET_GLOBAL_MNH, jit_fn_type($(mnh : int64; ctx : void?) : void? {}),
-        get_jit_get_global_mnh(), fixed_array(nounwind, willreturn), (nocapture, [1]))
+        get_jit_get_global_mnh(), fixed_array(nounwind, willreturn, memory_none), (nocapture, [1]))
     jit_add_extern(FN_JIT_GET_SHARED_MNH, jit_fn_type($(mnh : int64; ctx : void?) : void? {}),
-        get_jit_get_shared_mnh(), fixed_array(nounwind, willreturn), (nocapture, [1]))
+        get_jit_get_shared_mnh(), fixed_array(nounwind, willreturn, memory_none), (nocapture, [1]))
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLVM/daslib/llvm_jit_common.das:817]: Reject - the declared effect is the invariant the design rests on, not an overstatement. `tabGMnLookup` is built once in `Program::simulate` (`ast_simulate.cpp:3565`) and shared, never mutated, by every context cloned from that program (`context.cpp:250`, `:347`); the mangled-name hash is a constant of the program, and `context.globals` / `context.shared` are fixed for the context's lifetime. So for the lifetime of any context the JIT code runs on, the result is a pure function of `(mnh, ctx)`, which is exactly `memory(none)`. `memory(argmem: read)` would be the wrong effect for what it buys: the globals themselves are reachable through the `ctx` argument, so every store to a global would invalidate the lookup and a loop that writes one global would probe the hash on every iteration again - the shape this change removes. The call is not `speculatable`, so LLVM never hoists it past a branch (`modules/dasLLVM/ARCHITECTURE.md` sec.11 states the invariant and the consequence; `tests/llvm_jit_global_lookup.das` pins the emitted shape).

File: modules/dasLLVM/tests/llvm_exe_native_paths.das [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    }))
+}
+
+//! builds the fixture root with the given extra script args and runs the artifact; returns the run's stdout
+def private build_and_run(t : T?; dir : string; tag : string; extra_args : string) : string {
+    let root = "modules/dasLLVM/tests/_exe_native_paths_root.das"
+    let exe = path_join(dir, "{tag}.exe")   //! daslang -exe appends `.exe` to -output itself
+    var bout = ""
+    let brc = run_capture("\"{das_exe()}\" -jit -exe -output \"{path_join(dir, tag)}\" {root} {extra_args} 2>&1", bout)
+    t |> equal(0, brc, "the -exe build ({tag}) succeeds: {tail(bout)}")
```
[the remark this answers is not in the document]
Comment (borisbat) [reply modules/dasLLVM/tests/llvm_exe_native_paths.das:38]: Accepted, fixed in 975e3c666 - the build and the host-rows read go through `run_and_capture`'s argv (no shell), so the command has no leading quote to lose and a path with spaces needs no quoting; the exe run keeps its `set`/assignment prefix because it has to reach PATH.

File: tests/module_cache/test_descriptor_manifest.das [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+        tmp = tmp,
+        modDir = modDir,
+        dmDir = dmDir,
+        brokenDir = brokenDir,
+        descriptor = path_join(modDir, ".das_module"),
+        manifest = manifest,
+        manifestTmp = "{manifest}.tmp",
+        dmManifest = path_join(dmDir, ".das_module.manifest"),
+        cmd = "{das_exe()} -dasroot {get_das_root()} -no-module-cache -project_root {tmp} {path_join(tmp, "use.das")}"
+    )
```
[the remark this answers is not in the document]
Comment (borisbat) [reply tests/module_cache/test_descriptor_manifest.das:160]: Accepted, fixed in 975e3c666 - the child command opens with the `DAS_TRACE_MODULE_LOAD=1` assignment (`set "..."&&` on Windows) and quotes the binary and every path; the parent no longer sets and restores the variable in its own environment. The three sibling tests in this folder build their commands the same unquoted way and predate this PR; they are left as they are here.
